tests/arrays/test_base_string_array.py

```diff
@@ -0,0 +1,56 @@
+from collections import UserList
+from collections.abc import (
+    Callable,
+    Sequence,
+)
+from typing import (
+    TYPE_CHECKING,
+    Any,
+)
+
+import numpy as np
+import pandas as pd
+from pandas.core.arrays.string_ import BaseStringArray
+import pytest
+from typing_extensions import assert_type
+
+from tests import check
+from tests.utils import powerset
+
+
+@pytest.mark.parametrize("typ", [list, tuple, UserList])
+@pytest.mark.parametrize("data", powerset(["🐼", np.str_("🐼")], 1))
+@pytest.mark.parametrize("missing_values", powerset([np.nan, None, pd.NA]))
+def test_construction_sequence(
+    data: tuple[str | np.str_, ...],
+    missing_values: tuple[Any, ...],
+    typ: Callable[[Sequence[Any]], Sequence[Any]],
+) -> None:
+    check(pd.array(typ([*data, *missing_values])), BaseStringArray)
+
+    if TYPE_CHECKING:
+        assert_type(pd.array(["🐼", np.str_("🐼")]), BaseStringArray)
+
+        assert_type(pd.array(["🐼", np.str_("🐼"), None]), BaseStringArray)
+        assert_type(pd.array(["🐼", np.str_("🐼"), pd.NA]), BaseStringArray)
+
+        assert_type(pd.array(["🐼", np.str_("🐼"), None, pd.NA]), BaseStringArray)
+
+        assert_type(pd.array(("🐼", np.str_("🐼"))), BaseStringArray)
+        assert_type(pd.array(("🐼", np.str_("🐼"), pd.NA)), BaseStringArray)
+
+        assert_type(pd.array(UserList(["🐼", np.str_("🐼")])), BaseStringArray)
```

Comment:
can we just use a normal string like `"abc"` here and elsewhere instead of the unusual one?

Also, need to add a test like `pd.array(["abc", "def"])`